editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -2254,7 +2254,11 @@ void Node3DEditorViewport::_update_freelook(real_t delta) {
 	Vector3 forward;
 	if (navigation_scheme == FREELOOK_FULLY_AXIS_LOCKED) {
 		// Forward/backward keys will always go straight forward/backward, never moving on the Y axis.
-		forward = Vector3(0, 0, -1).rotated(Vector3(0, 1, 0), camera->get_rotation().y);
+		if (camera->get_rotation().x == float(-Math_PI / 2)) {
```

Comment:
Equality check on float values is not a good practice, as the result is always unreliable due to precision errors.

Reply:
I've tried `double(camera->get_rotation().x) == -Math_PI / 2.` although that doen't seem to work, and I'm not sure why.

Reply:
Precision errors. You should almost never check for equality between float or double (both are floating-point stored values). When you can, you should instead use `<=` or `>=`. Here is likely one of those cases.

Reply:
So something like `if (camera->get_rotation().x <= 1.5708)`?

Reply:
Simply `if (camera->get_rotation().x <= -Math_PI / 2)` I guess. Provided it's not supposed to go any lower I guess.

Reply:
d0279e17be22cb3e8a80dcf22e5decbc196f754a